chore(template): improve PullRequest template (#5622)
JanDeDobbeleer/oh-my-posh: .github/PULL_REQUEST_TEMPLATE.md

## Prerequisites

- [ ] I have read and understood the [Contributing Guide][contributing].
- [ ] My commit message follows the [Conventional Commits][cc] guidelines.

> - [ ] Relevant tests have been added or updated (for bug fixes/features).
> - [ ] Documentation has been added or updated (for bug fixes/features).

Reviewer: ```suggestion
- [ ] Relevant tests have been added or updated (for bug fixes or features).
- [ ] Documentation has been added or updated (for bug fixes or features).
```

Instead of separating phrases with a slash, I think using an "or" is clearer. Just a small suggestion. :)

## Prerequisites

- [ ] I have read and understood the [Contributing Guide][contributing].

> - [ ] My commit messages follow the [Conventional Commits][cc] guidelines and our [commitlint rules and standards][commitlint].

Reviewer 2: commitlint enforces conventional commits, so I find that phrase rather strange.

```suggestion
- [ ] My commit messages are written according the [Conventional Commits][cc] guidelines, respecting our [rules][commitlint].
```

- [ ] Relevant tests have been added or updated (for bug fixes or features).
- [ ] Documentations have been added or updated (for bug fixes or features).

## Description

<!--

**Description of Changes:**

> Please provide a clear and concise description of the changes introduced by this pull request. You should explain the following:

Reviewer 2: ```suggestion
Please provide a clear and concise description of the changes introduced by this pull request. Explain the following when relevant:
```

1. **Summary:** Briefly describe the main change or addition in this PR. What is the overall purpose of your changes?
2. **Problem:** Specify the issue or problem this PR addresses. Include any relevant context or background to explain why this change is necessary.
3. **Solution:** Explain how your changes resolve the issue or add the new feature. Provide key implementation details to clarify how the solution works.
4. **Impact:** Note any potential impacts or side effects of your changes. Mention if additional testing is needed or if there are any known limitations.
5. **Issue Closure:** Indicate which issues this PR resolves or closes.

**Example:**
"This PR introduces a new feature for X by implementing Y that will optimize the query logic, which reduces execution time and prevents the error. This change addresses the issue reported in #123.

Closes #123"

Please ensure that each section is thoroughly explained to assist reviewers in understanding your changes.

-->

---

> <br>

Reviewer 2: why are the linebreaks here?

Author: The line breaks are used to emphasize the main PR description section, making it easier to focus on providing a clear and concise overview of the changes.

<br>
<br>

<!-- _Description of your changes:_ -->

<br>
<br>
<br>
<br>

---

> ### Helpful Resources

Reviewer 2: there's a reason this was hidden, we shouldn't see this after creation

Author: Are you sure you want to comment that? It has a styling aspect for the preview, but it will also take up some space.

- **[Getting Started Guide][started-guide]:** For first-time contributors, this guide will help you get acquainted with the project.
- **[Oh My Posh Git Guide][git-guide]:** For understanding Git practices and commands.
- **[Discussions Page][discussions]:** Join project discussions and ask questions.
- **[GitKraken Guide][gitkraken]:** Recommended tool for managing Git repositories.

> <!-- - **[Code Structure Guide][code-structure]:** Understand the overall organization of our codebase. -->

Reviewer 2: this isn't relevant here

Author: I think it's somewhat related to the pull request. It helps contributors better understand the PR.
I think adding this section to the docs is a good idea; we can link to the page we’ll create in the template. We can use this interim solution until the docs section is finalized.

- **[Community Standard][community]:** Engage with the community and understand our community standards.
- **[Contributing Guide][contributing]:** Read and understand our guidelines for contributing.
- **[GitHub’s Pull Request Review Guide][gh-pr-review]:** Additional guidance on the review process.

> <!-- - **[Pitfalls to Avoid][pitfalls]:** Common mistakes to watch out for when contributing. -->

Reviewer 2: this isn't relevant here

Author: I think it's somewhat related to the pull request. It helps contributors better understand the PR.
I think adding this section to the docs is a good idea; we can link to the page we’ll create in the template. We can use this interim solution until the docs section is finalized.
